### PR TITLE
Fix follower tick

### LINKED ISSUE #56: security vulnr: follower tick not checking if stateRoot hash is correct
the follower tick is checking if the  `balance` tree in the newState msg represents a proper state transition, but it's not checking if the `stateRoot` is the correct hash for that tree

two solutions

1. check the `stateRoot` against our own computed stateRoot via `adapter.getSignableStateRoot(channel, tree.getRoot())`
2. compute that `stateRoot` at the end when signing; that way, even if the leader tricks us, we won't sign *their* stateRoot

**But before doing anything, we should first write an integration test that proves this**

### PR DESCRIPTION
**Pull request by samparsky**
Fixes #56 
it depends on the PR #55 being merged

### CODE REVIEW

File: services/validatorWorker/lib/followerRules.js [hunk 1]
```diff
@@ -47,4 +47,19 @@ function sumMins(our, approved) {
 	)
 }
 
-module.exports = { isValidTransition, isHealthy }
+function getRootHash(channel, balances, adapter){
```
Comment (Ivshti) [inline services/validatorWorker/lib/followerRules.js]: I think you can export that and reuse it in `leader.js`, right?

File: test/integration.js [hunk 2]
```diff
@@ -258,6 +258,38 @@ tape('POST /channel/{id}/{validator-messages}: wrong signature', function(t) {
 	.catch(err => t.fail(err))
 })
 
+tape('POST /channel/{id}/{validator-messages}: wrong root hash', function(t) {
```
Comment (Ivshti) [inline test/integration.js]: can you change that to "wrong (deceptive) root hash"

File: test/integration.js [hunk 3]
```diff
@@ -258,6 +258,38 @@ tape('POST /channel/{id}/{validator-messages}: wrong signature', function(t) {
...
+	const stateRoot = "6def5a300acb6fcaa0dab3a41e9d6457b5147a641e641380f8cc4bf5308b16f1"
+
+	fetch(`${followerUrl}/channel/${dummyVals.channel.id}/validator-messages`, {
```
Comment (Ivshti) [inline test/integration.js]: did you try if this test fails w/o the check?

Comment (samparsky) [reply test/integration.js]: Yes, it passes without the check and fails with the check

File: test/integration.js (hunk 2 above)
Comment (samparsky) [reply test/integration.js]: okay will do

File: services/validatorWorker/lib/followerRules.js (hunk 1 above)
Comment (samparsky) [reply services/validatorWorker/lib/followerRules.js]: yeah definitely

File: test/integration.js (hunk 2 above)
Comment (samparsky) [reply test/integration.js]: okay

File: test/integration.js (hunk 3 above)
Comment (samparsky) [reply test/integration.js]: yes I did, it passes without the check and fails with the check

**Pull request merged**